src/main/java/uk/ac/ic/wlgitbridge/snapshot/servermock/state/SnapshotAPIStateBuilder.java

```diff
@@ -83,16 +83,30 @@ private void addGetDocForProject(
             String projectName,
             JsonObject jsonGetDoc
     ) {
-        getDoc.put(
-                projectName,
-                new GetDocResult(
-                        jsonGetDoc.get("error"),
-                        jsonGetDoc.get("versionID").getAsInt(),
-                        jsonGetDoc.get("createdAt").getAsString(),
-                        jsonGetDoc.get("email").getAsString(),
-                        jsonGetDoc.get("name").getAsString()
-                )
-        );
+        if (jsonGetDoc.has("migratedFromId")) {
```

Comment:
It seems like quite a lot of duplication to call different constructors. What do you think to something like
```
String migratedFromId = null;
if (jsonGetDoc.has("migratedFromId")) {
  migratedFromId = jsonGetDoc.get("migratedFromId")
}
```
and then pass that to the new constructor.

Also to check: do we still need the old constructor without this argument?

Reply:
Yup, turns out we can just use the one constructor with a null at the end :+1: